packages/src/components/editable/editable.component.ts

```diff
@@ -478,185 +589,386 @@ export class SlateEditableComponent implements OnInit, OnChanges, OnDestroy, Aft
+                // triggering a `beforeinput` expecting the change to be applied to the immediately before
+                // set selection.
+                this.scheduleOnDOMSelectionChange.flush();
+                this.onDOMSelectionChange.flush();
+        
                 const { selection } = editor;
                 const { inputType: type } = event;
-                const data = event.dataTransfer || event.data || undefined;
-                event.preventDefault();
-
+                const data = (event as any).dataTransfer || event.data || undefined;
+        
+                const isCompositionChange =
+                    type === "insertCompositionText" || type === "deleteCompositionText";
+        
+                // COMPAT: use composition change events as a hint to where we should insert
+                // composition text if we aren't composing to work around https://github.com/ianstormtaylor/slate/issues/5038
+                if (isCompositionChange && AngularEditor.isComposing(editor)) {
+                    return;
+                }
+        
+                let native = false;
+                if (
+                    type === "insertText" &&
+                    selection &&
+                    Range.isCollapsed(selection) &&
+                    // Only use native character insertion for single characters a-z or space for now.
+                    // Long-press events (hold a + press 4 = ä) to choose a special character otherwise
+                    // causes duplicate inserts.
+                    event.data &&
+                    event.data.length === 1 &&
+                    /[a-z ]/i.test(event.data) &&
+                    // Chrome has issues correctly editing the start of nodes: https://bugs.chromium.org/p/chromium/issues/detail?id=1249405
+                    // When there is an inline element, e.g. a link, and you select
+                    // right after it (the start of the next node).
+                    selection.anchor.offset !== 0
+                ) {
+                    native = true;
+        
+                    // Skip native if there are marks, as
+                    // `insertText` will insert a node, not just text.
+                    if (editor.marks) {
+                        native = false;
+                    }
+        
+                    // Chrome also has issues correctly editing the end of anchor elements: https://bugs.chromium.org/p/chromium/issues/detail?id=1259100
+                    // Therefore we don't allow native events to insert text at the end of anchor nodes.
+                    const { anchor } = selection;
+        
+                    const [node, offset] = AngularEditor.toDOMPoint(editor, anchor);
+                    const anchorNode = node.parentElement?.closest("a");
+        
+                    const window = AngularEditor.getWindow(editor);
+        
+                    if (
```

Comment:
There's already a PR on the spellcheck blinking issue, but I'm not sure if it should be merged on the master branch.

Reply:
https://github.com/worktile/slate-angular/pull/197